contracts/utils/structs/BitMaps.sol

```diff
@@ -33,10 +33,12 @@ library BitMaps {
      * @dev Sets the bit at `index` to the boolean `value`.
      */
     function setTo(BitMap storage bitmap, uint256 index, bool value) internal {
-        if (value) {
-            set(bitmap, index);
-        } else {
-            unset(bitmap, index);
+        uint256 bucket = index >> 8;
+        uint256 mask = 1 << (index & 0xff);
+        bool old = bitmap._data[bucket] & mask != 0;
+
+        if (old != value) {
+            bitmap._data[bucket] ^= mask;
```

Comment:
We should avoid duplicating sload.
```suggestion
        uint256 data = bitmap._data[bucket];
        bool oldValue = data & mask != 0;

        if (oldValue != value) {
            bitmap._data[bucket] = data ^ mask;
        }
```

Reply:
Yes, I will resubmit this PR today later.

Reply:
Updated !